Feat: autocomplete searchbox (#83)
dlw-digitalworkplace/dw-component-lib: packages/dw-react-controls/src/components/AutocompleteSearchBox/stories/AutocompleteSearchBox.stories.mdx

import { ArgsTable, Meta, Story, Canvas } from "@storybook/addon-docs";
import { AutocompleteSearchBox } from "../AutocompleteSearchBox";
import * as stories from "./AutocompleteSearchBox.stories";

<Meta title="Components/AutocompleteSearchBox" component={AutocompleteSearchBox} />

# AutocompleteSearchBox

The autocomplete searchbox is a component that expands on the Fluent UI React Searchox with an autocomplete dropdown.

> The component is inspired and build upon the Microsoft example: [https://github.com/microsoft/autocomplete-search-box]()

Reviewer: ```suggestion
The component is inspired and built upon the Microsoft example: [https://github.com/microsoft/autocomplete-search-box]()
```

> When typing the searchbox, autocomplete suggestions are dynamically fetched and displayed in a dropdown below the searchbox.

Reviewer: ```suggestion
When typing in the searchbox, autocomplete suggestions are dynamically fetched and displayed in a dropdown below the searchbox.
```

### Basic usage

This example shows the basic usage of the AutocompleteSearchBox component.

> The only required property is an `onResolveSuggestions` funtion that return a promise with the suggestions.

Reviewer: ```suggestion
The only required property is an `onResolveSuggestions` function that returns a promise with the suggestions.
```

<Canvas withToolbar={true}>
	<Story story={stories.Basic} name="Basic usage" />
</Canvas>

### Advance usage

> This example shows the advance usage of the AutocompleteSearchBox component.

Reviewer: ```suggestion
This example shows the advanced usage of the AutocompleteSearchBox component.
```

It is possible to control the value outside of the component by using the `value` and `onChange` properties.
It also shows some examples of the debounce functionality and how to use the `onSearch` event.

> There is also an optoin to disable the suggestion highlighting.

Reviewer: ```suggestion
There is also an option to disable the suggestion highlighting.
```
